group_project_v2/components/project_navigator.py

```diff
@@ -0,0 +1,402 @@
+                stage_state = self.get_stage_state(activity.scope_ids.usage_id, stage)
+                stages_data.append({'stage': stage, 'state': stage_state})
+
+            navigation_map.append({
+                'id': activity.scope_ids.usage_id,
+                'display_name': activity.display_name,
+                'stages': stages_data
+            })
+
+        context = {'view': self, 'navigation_map': navigation_map}
+        return self.render_student_view(context)
+
+
+class ResourcesViewXBlock(ProjectNavigatorViewXBlockBase):
+    type = ViewTypes.RESOURCES
+    icon = u"fa-files-o"
+    display_name_with_default = _(u"Resources")
+
+    template = "resources_view.html"
+    css_file = "resources_view.css"
+    js_file = "resources_view.js"
+    initialize_js_function = "GroupProjectNavigatorResourcesView"
+
+    def student_view(self, context):  # pylint: disable=unused-argument
+        resources_map = []
+        for activity in self.navigator.group_project.activities:
+            resources = list(itertools.chain(*[
+                stage.resources for stage in activity.get_group_activity().activity_stages
+            ]))
+            resources_map.append({
+                'id': activity.scope_ids.usage_id,
+                'display_name': activity.display_name,
+                'resources': resources,
+            })
+
+        context = {'view': self, 'resources_map': resources_map}
+        return self.render_student_view(context)
+
+
+# pylint-disable=no-init
+@XBlock.needs('user')
+@XBlock.wants('notifications')
+class SubmissionsViewXBlock(ProjectNavigatorViewXBlockBase):
+    type = ViewTypes.SUBMISSIONS
+    icon = u"fa-upload"
+    display_name_with_default = _(u"Submissions")
+
+    template = "submissions_view.html"
+    css_file = "submissions_view.css"
+    js_file = "submissions_view.js"
+    initialize_js_function = "GroupProjectNavigatorSubmissionsView"
+
+    def _get_submissions_map(self):
+        submissions_map = []
+        for activity in self.navigator.group_project.activities:
+            group_activity = activity.get_group_activity()
+            group_activity.update_submission_data(
+                project_api.get_latest_workgroup_submissions_by_id(activity.workgroup["id"])
+            )
+            stages = [stage for stage in group_activity.activity_stages if stage.submissions_stage]
+            submissions_required = any(True for stage in stages if len(stage.submissions) > 0)
+            submissions_map.append({
+                'id': activity.scope_ids.usage_id,
+                'display_name': activity.display_name,
+                'submissions_required': submissions_required,
+                'stages': stages,
+            })
+
+        return submissions_map
+
+    def student_view(self, context):  # pylint: disable=unused-argument
+        # FIXME: should have used `include` in template, but it can't find the template: likely resource loader does
```

Comment:
@antoviaque if I do understand right xblock-utils that are used to render templates already uses Django templates and have dependencies on Django:

* `from django.template import Context, Template` in xblockutils/resources.py
* `from django.utils.translation import ugettext` in xblockutils/studio_editable.py

If I do understand right, the idea is that XBlocks should not *assume* some specific template engine or presense of Django - instead they should specify the dependencies implicitly. XBlocks definitely responsible for rendering themselves, and doing so definitely requires *some* template engine.

Anyway, adding support for include needs to be done in xblock-utils, so it might be a good idea to raise a question with edX:
* should xblock-utils still use Django templates and if so, it might need some improvement (as it does not support some features)
* or should it switch to some other template engine and explicitly mention it in requirements.